src/pycodestyle/plugins.rs

```diff
@@ -0,0 +1,203 @@
+use fnv::FnvHashMap;
+use itertools::izip;
+use rustpython_parser::ast::{Cmpop, Constant, Expr, ExprKind};
+
+use crate::ast::types::Range;
+use crate::autofix::Fix;
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckKind, RejectedCmpop};
+use crate::code_gen::SourceGenerator;
+
+fn compare(left: &Expr, ops: &[Cmpop], comparators: &[Expr]) -> Option<String> {
+    let cmp = Expr::new(
+        Default::default(),
+        Default::default(),
+        ExprKind::Compare {
+            left: Box::new(left.clone()),
+            ops: ops.to_vec(),
+            comparators: comparators.to_vec(),
+        },
+    );
+    let mut generator = SourceGenerator::new();
+    if let Ok(()) = generator.unparse_expr(&cmp, 0) {
+        if let Ok(content) = generator.generate() {
+            return Some(content);
+        }
+    }
+    None
+}
+
+/// E711, E712
+pub fn literal_comparisons(
+    checker: &mut Checker,
+    expr: &Expr,
+    left: &Expr,
+    ops: &[Cmpop],
+    comparators: &[Expr],
+    check_none_comparisons: bool,
+    check_true_false_comparisons: bool,
+) {
+    // Mapping from (bad operator index) to (replacement operator). As we iterate
+    // through the list of operators, we apply "dummy" fixes for each error,
+    // then replace the entire expression at the end with one "real" fix, to
+    // avoid conflicts.
```

Comment:
@harupy - The one issue with this approach is that if we start to show users fix suggestions in the `ruff` output (similar to Clippy), we'll have to be careful in how we handle this case.